lib/glific/messages.ex

```diff
@@ -954,6 +960,24 @@ defmodule Glific.Messages do
     end)
   end
 
+  # apply filter for message labels
+  @spec include_label_filter(Ecto.Queryable.t(), []) :: Ecto.Queryable.t()
+  defp include_label_filter(query, []), do: query
+
+  defp include_label_filter(query, label_ids) do
+    flow_labels =
+      Glific.Flows.FlowLabel
+      |> where([f], f.id in ^label_ids)
+      |> select([f], f.name)
+      |> Repo.all()
+
+    flow_labels
+    |> Enum.reduce(query, fn flow_label, query ->
+      where(query, [c], ilike(c.flow_label, ^"%#{flow_label}%"))
+    end)
+    |> or_where([m], m.flow_label in ^flow_labels)
```

Comment:
Is this working with multiple flows for one message?

Reply:
yes